File: sdk/java/apps/sample/src/main/java/io/touca/apps/sample/StudentsTest.java
// Copyright 2022 Touca, Inc. Subject to Apache-2.0 License.

package io.touca.apps.sample;

import io.touca.Touca;
import java.time.LocalDate;

public final class StudentsTest {

  @Touca.Workflow
  public void findStudent(final String username) {
    Touca.startTimer("find_student");
    Student student = Students.findStudent(username);
    Touca.stopTimer("find_student");
    Touca.assume("username", student.username);
    Touca.check("fullname", student.fullname);
    Touca.check("birth_date", student.dob);
    Touca.check("gpa", student.gpa);
    Touca.addMetric("external_source", 1500);
  }

  public static void main(String[] args) {
    Touca.addTypeAdapter(LocalDate.class, x -> x.toString());
    Touca.setWorkflowOptions("findStudent", x -> {
      x.testcases = new String[] { "alice", "bob", "charlie" };
    });
    Touca.run(StudentsTest.class, args);
  }
}

File: sdk/cpp/tests/sample_app/students_test.cpp
// Copyright 2023 Touca, Inc. Subject to Apache-2.0 License.

#include "students.hpp"

#include "students_types.hpp"
#include "touca/touca.hpp"

int main(int argc, char* argv[]) {
  touca::workflow(
      "students",
      [](const std::string& username) {
        const auto& student = find_student(username);
        touca::assume("username", student.username);
        touca::check("fullname", student.fullname);
        touca::check("birth_date", student.dob);
        touca::check("gpa", student.gpa);
        touca::add_metric("external_source", 1500);
      },
      [](touca::WorkflowOptions& options) {
        options.testcases =
            std::vector<std::string>({"alice", "bob", "charlie"});
      });
  return touca::run(argc, argv);
}
